controllers/databasecluster_controller.go

```diff
@@ -955,16 +955,13 @@ func (r *DatabaseClusterReconciler) genPXCHAProxySpec(
 	haProxy := r.defaultPXCSpec().HAProxy
 	// Set configuration based on database size.
 	if database.Spec.Engine.Resources.Memory.Cmp(memoryLargeSize) == 0 {
-		haProxy.PodSpec.Resources = haProxyResourceRequirementsLarge
 		haProxy.PodSpec.LivenessProbes.TimeoutSeconds = 30
 		haProxy.PodSpec.ReadinessProbes.TimeoutSeconds = 30
 	}
-	if database.Spec.Engine.Resources.Memory.Cmp(memoryMediumSize) == 0 {
-		haProxy.PodSpec.Resources = haProxyResourceRequirementsMedium
-	}
-	if database.Spec.Engine.Resources.Memory.Cmp(memorySmallSize) == 0 {
-		haProxy.PodSpec.Resources = haProxyResourceRequirementsSmall
-	}
+	//nolint:godox
+	// TODO (EVEREST-824): We need to figure out the correct values
```

Comment:
@mayankshah1607 I created https://perconadev.atlassian.net/browse/EVEREST-941 for this purpose please update the ticket number in this comment to EVEREST-941